packages/core/src/useNavigationHelpers.tsx

```diff
@@ -82,4 +94,27 @@ export default function useNavigationHelpers<
     setState,
     onAction,
   ]);
+
+  const { addNavigationListener } = React.useContext(NavigationFocusedContext);
+  const listener = React.useRef((callback: FocusedNavigationCallback) => {
+    if (navigation.isFocused()) {
```

Comment:
I was wondering if the `target` approach we following in `onAction` is better than checking `navigation.isFocused` as `isFocused` will check focused state of each parent. but lets leave for separate PR

Reply:
no strong opinion